### PR TITLE
Fix handling node state

### PR DESCRIPTION
**Pull request by marcelveldt**
This fixes 2 issues:

- On "node ready" event a full state of the node is dumped including values. The new values were not stored and thus HA discovery was failing when a new node was added to the mesh.
- When a node is in the process of being added, first a partial node state is sent, missing most values. Once the interview progresses/completes these values come in. So basically all attributes for the Node model are optional to account for this behaviour.

Fixes HA issue: https://github.com/home-assistant/core/issues/45631

### CODE REVIEW

File: zwave_js_server/model/device_class.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 from typing import List, Optional, TypedDict
 
 
-class DeviceClassDataType(TypedDict):
+class DeviceClassDataType(TypedDict, total=False):
```
Comment (cgarwood) [inline zwave_js_server/model/device_class.py:10]: Where is `total` being used?

Comment (marcelveldt) [reply zwave_js_server/model/device_class.py:10]: mypy didn't like the empty dict coming in...
if the node is being added this info is added at a later stage

Comment (MartinHjelmare) [reply zwave_js_server/model/device_class.py:10]: `total` is part of the TypedDict api in standard library.

File: zwave_js_server/model/node.py [hunk 2]
```diff
@@ -104,57 +104,57 @@ def ready(self) -> Optional[bool]:
...
     def device_class(self) -> DeviceClass:
         """Return the device_class."""
-        return DeviceClass(self.data["deviceClass"])
+        return DeviceClass(self.data.get("deviceClass", {}))
 
     @property
-    def is_listening(self) -> bool:
+    def is_listening(self) -> Optional[bool]:
         """Return the is_listening."""
         return self.data["isListening"]
```
Comment (MartinHjelmare) [inline zwave_js_server/model/node.py:112]: If the item is optional we need to use dict.get.

**Comment by MartinHjelmare** [comment]
It would be good to add tests for these cases mentioned in the description.

File: zwave_js_server/model/node.py (hunk 2 above)
Comment (marcelveldt) [reply zwave_js_server/model/node.py:112]: Oh I forgot one ;-)

**Comment by marcelveldt** [comment]
> It would be good to add tests for these cases mentioned in the description.

How? The info is completely random. Sometimes values XYZ included, sometimes not.
We have to assume a dict with node_id as bare minimum and more info comes in later.

**Comment by MartinHjelmare** [comment]
We simulate the conditions mentioned above.

**Comment by marcelveldt** [comment]
> We simulate the conditions mentioned above and make sure the library acts appropriately.

OK, again, how ? I have no idea. You're the tests wizard ;-)

**Comment by marcelveldt** [comment]
this is what happens btw:
- node inclusion started
- node added event with limited node object (no values and not all keys there)
- node read event (all/most keys there and values list)

**Comment by MartinHjelmare** [comment]
Yes, exactly. Create a node with some minimum state. Then receive the node ready event with needed data and assert that values are updated ok.

**Comment by marcelveldt** [comment]
test added

**Review by MartinHjelmare — APPROVED** [review]
Looks good!

**Pull request merged**